### PR TITLE
Refactor drop maps

### PR DESCRIPTION
**Pull request by eindacor**

### CODE REVIEW

**Comment by rDarge** [comment]
It would have been nice to see the functional changes separated out from all of the changes to error logging... 

🍝/ 10

**Comment by eindacor** [comment]
TBH I'd never done a PR on github before, just wanted to see how it worked, but I agree.

edit: alright, that's better

File: lib/MapCacheIF.js [hunk 1]
```diff
@@ -0,0 +1,215 @@
+MapCacheIF = function(value_map) {
```
Comment (eindacor) [inline lib/MapCacheIF.js:1]: This should be separated into two different classes. One for int maps, one for float maps

Comment (eindacor) [reply lib/MapCacheIF.js:1]: and for the love of god, clean this thing up

File: lib/global_variables.js [hunk 2]
```diff
@@ -59,7 +59,7 @@ ORIGINAL_VALUE_BUFF = 7;
 ITEM_LEVEL_VALUE_BUFF = .01;
 
 DYNAMIC_CRATE_DISCOUNT_COEFFICIENT = .95;
-DYNAMIC_CRATE_COUNT = 4;
+DYNAMIC_ITEMS_PER_CRATE = 4;
```
Comment (eindacor) [inline lib/global_variables.js:61]: accidental var rename

File: client/templates/adminDashboard/adminDashboard.html [hunk 3]
```diff
@@ -6,19 +6,7 @@
 				<h1 class="af-color">Admin Tools</h1>
 				<div class="row tools-section">
 					<h4 class="af-color">Drop Data</h4>
-					<canvas id="drop-chart" width="1000px" height="600px">{{updateChart}}</canvas>
-					<table id="rarity-map-table">
-					{{#each ARTWORK_RARITIES}}
```
Comment (showcasefloyd) [inline client/templates/adminDashboard/adminDashboard.html:11]: Tables ? We will work on this for sure. :)

**Comment by showcasefloyd** [comment]
What? a pull request !?! Now this officially feels like work :) BTW is there some approve button I'm supposed to click here?

File: client/templates/adminDashboard/adminDashboard.html (hunk 3 above)
Comment (eindacor) [reply client/templates/adminDashboard/adminDashboard.html:11]: Yeah but any wonky front-end stuff in the code is now officially your fault. That's how legacy code works.

File: lib/MapCacheIF.js (hunk 1 above)
Comment (rDarge) [reply lib/MapCacheIF.js:1]: no u

File: client/templates/adminDashboard/adminDashboard.html (hunk 3 above)
Comment (rDarge) [reply client/templates/adminDashboard/adminDashboard.html:11]: table haters >:(

File: server/npc_interactions/historianInteraction.js [hunk 4]
```diff
@@ -20,7 +20,7 @@ var generateTarget = function(default_target_count) {
 }
 
 var generateQuest = function(rarity, is_own_gallery, player_interface) {
-    var money = getAverageDropValue(Meteor.user().profile.level, 0) * 10;
+    var money = getAverageDropValue(Meteor.user().profile.level, 0) * 6;
```
Comment (rDarge) [inline server/npc_interactions/historianInteraction.js]: It seems like there are a lot of arbitrary constants kicking around related to rewards in a bunch of different server files... would it make sense to define constants in one place that could be referenced across different contexts? Or at least at the top of the .js file so you don't have to hunt through various function bodies to find them again. just food for thought from the peanut gallery

Comment (eindacor) [reply server/npc_interactions/historianInteraction.js]: Yes, absolutely. These should be centralized.

File: lib/MapCacheIF.js (hunk 1 above)
Comment (eindacor) [reply lib/MapCacheIF.js:1]: too late, i already assigned it to you on the trello board

**Pull request merged**